src/main/java/michal/kruczala/project/ptt/ParserPTT.java

```diff
@@ -3,45 +3,58 @@
 import michal.kruczala.project.ptt.readers.WebReader;
 
 import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import lombok.extern.slf4j.Slf4j;
 
+
 @Slf4j
 public class ParserPTT {
     private static final Logger LOGGER = LoggerFactory.getLogger(ParserPTT.class);
     private static final WebReader webReader = new WebReader();
-    private static final CompetitionNumberParser lookForCompNumbersObject = new CompetitionNumberParser();
-    private static final LookForArchivalYearComps lookForArchivalYearComps = new LookForArchivalYearComps();
+    static final CompetitionNumberParser competitionNumberParser = new CompetitionNumberParser();
+    private static final ArchivalCompetitionParser lookForArchivalYearComps = new ArchivalCompetitionParser();
+    private static final CompetitionSitesParser competitionSitesParser = new CompetitionSitesParser();
     private static final String PTTWebSite = "https://baza.taniec.pl/?v=turnieje&p=arch";
 
-
     public static void main(String[] args) throws IOException {
 
-        String result = webReader.downloadWebPage(PTTWebSite);
-        LOGGER.debug(result);
+        String pageContent = webReader.downloadWebPage(PTTWebSite);
+        LOGGER.debug(pageContent);
 
-        extractCompNumbersFromPTTWebsite(result);
-        LOGGER.debug(extractCompNumbersFromPTTWebsite(result));
+        Set<String> competitionNumbers = competitionNumberParser.parse(pageContent);
+        LOGGER.debug(String.valueOf(competitionNumbers));
 
 
-        displayArchivalPttWebsite();
-        LOGGER.debug(displayArchivalPttWebsite());
+        String chosenArchivalPttWebSite = webReader.downloadWebPage(PTTWebSite + lookForArchivalYearComps.yearChosenByUser());
+        LOGGER.debug(chosenArchivalPttWebSite);
 
-    }
 
-    private static String displayArchivalPttWebsite() throws IOException {
-        return String.valueOf(webReader.downloadWebPage(PTTWebSite + lookForArchivalYearComps.lookForArchivalYearCompsENDURL()));
-    }
+        List<String> ListOfAllCompetitionsLinks2022 = competitionSitesParser.getListOfCompetitionSites(pageContent);
+        LOGGER.debug(String.valueOf(ListOfAllCompetitionsLinks2022));
+
+
+        contentFromAllCompetitionSites2022(ListOfAllCompetitionsLinks2022);
 
-    private static String extractCompNumbersFromPTTWebsite(String result) {
-        return String.valueOf(lookForCompNumbersObject.lookForCompetitionNumbers(result));
     }
 
-}
 
+    private static void contentFromAllCompetitionSites2022(List<String> ListOfAllCompetitionsLinks2022) {
```

Comment:
2022 nie jest rokiem końcowym, bieżący rok jest końcowym, w przyszłym roku to będzie 2023

Reply:
Zrobione :)